packages/cubejs-client-core/src/time.ts

```diff
@@ -202,6 +183,65 @@ function alignToOrigin(startDate: dayjs.Dayjs, interval: ParsedInterval, origin:
   return alignedDate;
 }
 
+export const dayRange = (from: any, to: any, annotations?: Record<string, { granularity?: Granularity }>): DayRange => ({
+  by: (value: any) => {
+    const results = [];
+
+    let start = internalDayjs(from);
+    const end = internalDayjs(to);
+
+    while (start.startOf(value).isBefore(end) || start.isSame(end)) {
+      results.push(start);
+      start = start.add(1, value);
```

Comment:
How would this work for custom intervals like `3 days` or `2 days 3 hours 4 minutes`?